submissions/ustymchyk/js-post-oop/index.js

```diff
@@ -118,64 +120,101 @@ class Human extends Creature {
+        let children = this.children.length ? `My ` + this.children.length > 1 ? `children: ${this._getChildren()}` : `child: ${this._getChildren()}` : '';
 
-        if (this.#friends.length) {
-            friends = `There is my friends: ${this.getFriendsNames()}. `;
-        }
+        return intro + friends + spouse + children;
+    }
+
+    bornCreature(partner, femaleName, maleName) {
+        if (this.spouse === partner) {
+            const child = super.bornCreature(partner, femaleName, maleName);
+
+            this.children.push(child);
+            partner.addChild(child);
 
-        if (this.spouse) {
-            spouse = `And i have a spouse ${this.spouse}; `;
+            return child;
         }
+    }
 
-        return intro + friends + spouse;
+    addChild(child) {
+        this.children.push(child);
+    }
 
+    _getChildren() {
+        return this.children.map(child => child.name).join('; ');
     }
 
     _createContact(otherHuman) {
-        const name = otherHuman.name;
         const random = Math.random();
 
-        this.#contacts.push(name);
+        this.#contacts.push(otherHuman);
 
         if (random > .3) {
-            this.#friends.push(name);
+            this.#friends.push(otherHuman);
         }
 
         if (otherHuman.gender !== this.gender && random > .5) {
-            this.#sympathy.push(otherHuman.name);
+            this.#sympathy.push(otherHuman);
         }
     }
 }
 
+class Names {
+    #maleNames = [
+        'Liam', 'Noah', 'Oliver', 'William', 'Elijah', 'James', 'Benjamin',
+        'Mason', 'Ethan', 'Alexander', 'Henry', 'Jacob', 'Michael', 'Daniel',
+        'Logan', 'Jackson', 'Sebastian', 'Jack', 'Aiden', 'Owen', 'Samuel', 'Matthew'];
+
+    #femaleNames = [
+        'Olivia', 'Emma', 'Ava', 'Sophia', 'Isabella', 'Charlotte', 'Amelia', 'Mia', 'Harper',
+        'Evelyn', 'Abigail', 'Emily', 'Ella', 'Elizabeth', 'Camila', 'Luna', 'Sofia', 'Avery',
+        'Mila', 'Aria', 'Scarlett', 'Penelope', 'Layla', 'Chloe', 'Victoria', 'Madison', 'Eleanor',
+        'Grace', 'Nora', 'Riley'];
+
+    getMaleName() {
+        return this._getRandomName(this.#maleNames);
+    }
+
+    getFemaleName() {
+        return this._getRandomName(this.#femaleNames);
+    }
+
+    _getRandomName(namesArr) {
+        return namesArr[Math.floor(Math.random() * namesArr.length)];
+    }
+}
+
 const getRandomAge = () => Math.floor(Math.random() * 15 + 20);
+const names = new Names();
 
-const men = ['Liam', 'Noah', 'Oliver', 'William', 'Elijah', 'James', 'Benjamin'].map(name => new Human(name, 'm', getRandomAge()));
-const women = ['Olivia', 'Emma', 'Ava', 'Sophia', 'Isabella', 'Charlotte', 'Amelia'].map(name => new Human(name, 'f', getRandomAge()));
+const men = [0, 1, 2, 3, 4].map(() => new Human(names.getMaleName(), 'm', getRandomAge()));
+const women = [0, 1, 2, 3, 4].map(() => new Human(names.getFemaleName(), 'f', getRandomAge()));
```

Comment:
Not a really valid use case for `.map`. We do not transform source array into new array, but rather use it as a quesi-iterator. What if we create dozens if not thousands of men and women? This approach won't scale up.
**You are not required to fix this now.**

Reply:
Totally agree. Just I`m enjoying how it looks in one line.  😂